metricbeat/module/windows/perfmon/pdh_integration_windows_test.go

```diff
@@ -336,3 +336,62 @@ func TestWildcardQuery(t *testing.T) {
 
 	t.Log(values)
 }
+
+func TestGroupByInstance(t *testing.T) {
+	config := Config{
+		CounterConfig:     make([]CounterConfig, 3),
+		GroupMeasurements: true,
+	}
+	config.CounterConfig[0].InstanceLabel = "processor.name"
+	config.CounterConfig[0].MeasurementLabel = "processor.time.pct"
+	config.CounterConfig[0].Query = `\Processor Information(_Total)\% Processor Time`
+	config.CounterConfig[0].Format = "float"
+
+	config.CounterConfig[1].InstanceLabel = "disk.bytes.name"
+	config.CounterConfig[1].MeasurementLabel = "disk.bytes.read.total"
+	config.CounterConfig[1].Query = `\FileSystem Disk Activity(_Total)\FileSystem Bytes Read`
+	config.CounterConfig[1].Format = "float"
+
+	config.CounterConfig[2].InstanceLabel = "processor.name"
+	config.CounterConfig[2].MeasurementLabel = "processor.time.idle.average.ns"
+	config.CounterConfig[2].Query = `\Processor Information(_Total)\Average Idle Time`
+	config.CounterConfig[2].Format = "float"
+
+	handle, err := NewPerfmonReader(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer handle.query.Close()
+
+	values, _ := handle.Read()
+
+	time.Sleep(time.Millisecond * 1000)
+
+	values, err = handle.Read()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.EqualValues(t, 1, len(values)) // Assert all metrics have been grouped into a single event
+
+	// Test all keys exist in the event
+	pctKey, err := values[0].MetricSetFields.HasKey("processor.time.pct")
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.True(t, pctKey)
+
+	pctKey, err := values[0].MetricSetFields.HasKey("disk.bytes.read.total")
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.True(t, pctKey)
+
+	pctKey, err := values[0].MetricSetFields.HasKey("processor.time.idle.average.ns")
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.True(t, pctKey)
+
+	t.Log(values)
```

Comment:
@smi9cm I tested this and it's producing more than one event. There are a few compilation errors in the test code so we are going to revert this the merge in #11001. I'd love it if you could reopen the PR after you have looked into the failing test case. Thanks!

```
    pdh_integration_windows_test.go:378: {
          "processor": {
            "name": "_Total",
            "time": {
              "idle": {
                "average": {
                  "ns": 10932693.47826087
                }
              },
              "pct": 0
            }
          }
        }
    pdh_integration_windows_test.go:378: {
          "disk": {
            "bytes": {
              "read": {
                "total": 0
              }
            }
          }
        }
```

Reply:
@smi9cm, Me and @jsoriano discussed the test case and made some changes and opened up a PR https://github.com/elastic/beats/pull/11002.

Reply:
New PR looks good to me, apologies for breaking the build there! Thanks @jsoriano as well.